tests/discover/filtering.sh

```diff
@@ -86,7 +88,7 @@ rlJournalStart
         rlAssertEquals "Check that number of fmf-ids equals to tests number" \
                        "$ids_amount" "$tests_amount"
         rlAssertEquals "Check url" "$url_discover" "$url_fmf_id"
-        rlAssertEquals "Check ref" "HEAD" "$ref_fmf_id"
+        rlAssertEquals "Check ref" "$expected_ref" "$ref_fmf_id"
```

Comment:
Reference about this change:
<details><summary>Previous output</summary>

```console
$ tmt run -r plan --name fmf/url/ref/path discover -h fmf --fmf-id finish | tee output
/var/tmp/tmt/run-012

/plans/fmf/url/ref/path
    discover
        how: fmf
        url: https://github.com/teemtee/tmt
        ref: eae4d52
        path: /examples/together
---
url: https://github.com/teemtee/tmt
ref: HEAD
path: /examples/together
name: /tests/full
---
url: https://github.com/teemtee/tmt
ref: HEAD
path: /examples/together
name: /tests/smoke
```

</details>

<details><summary>New output</summary>

```console
$ tmt run -r plan --name fmf/url/ref/path discover -h fmf --fmf-id finish | tee output
/var/tmp/tmt/run-012

/plans/fmf/url/ref/path
    discover
        how: fmf
        url: https://github.com/teemtee/tmt
        ref: eae4d52
        path: /examples/together
---
url: https://github.com/teemtee/tmt
ref: 0.10-8-geae4d527
path: /examples/together
name: /tests/full
---
url: https://github.com/teemtee/tmt
ref: 0.10-8-geae4d527
path: /examples/together
name: /tests/smoke
```

</details>

The `ref: HEAD` -> `ref: $(git describe --tags HEAD)` seems like an unexpected improvement to me :thinking:

Reply:
definitely!